Update roadmap after 0.3.0 release (#483)
eclipse-pass/main: docs/roadmap.md

If you are interested in helping to define and/or contribute to this roadmap, please get in touch. We're always happy to welcome new project contributors!

> # [Completed Releases](dev/CHANGELOG.md)

Reviewer: ```suggestion
# [Completed Releases](release-notes.md)
```
The CHANGELOG file was renamed to release-notes.md in #485

> # Completed Releases

Reviewer 2: Doesn't need to be changed, but what are your thoughts on moving the "completed releases" section to the end, so that the "upcoming (planned) releases" appears closer to the top of the document, since this just rehashes the information from the CHANGELOG doc?

Or do we need a section for completed releases, since the roadmap is a forward looking document?

Reviewer: The "Completed Releases" header could just become a link to the CHANGELOG (and the list of completed releases removed).

Author: Done

## [0.1.0](https://github.com/eclipse-pass/main/releases/tag/0.1.0)
## [0.2.0](https://github.com/eclipse-pass/main/releases/tag/0.2.0)
## [0.3.0](https://github.com/eclipse-pass/main/releases/tag/0.3.0)

# Planned Releases

## 0.4.0

> This release will focus on making the REST API feature complete and updating pass-ui

Reviewer: ```suggestion
This release will focus on making the REST API feature complete and connecting the UI to the completed API
```

* Access control for REST API

> * Replacement of all Golang web services.

Reviewer 2: This supersedes the line at https://github.com/eclipse-pass/main/blob/357a9cc0d0b21729822fdae8998fcf897e9c7d13/docs/roadmap.md?plain=1#L40, though granted, anything >1.0 is from the initial roadmap

Reviewer 2: I think I selected the wrong line for this comment, should be attached to Line 19:

> `* Replacement of all Golang web services`

Author: Got it.

* Remove mocking of the REST API from the user interface

## 0.5.0
This release will focus on tools to load data

> * Data loaders ported to new framework

Reviewer: ```suggestion
* Data loaders ported to utilize the new REST API and backend framework
```
